### DISCUSSION TITLE
Calculations seems to be little off

### DISCUSSION THREAD

**Discussion by HemKal** [gen-so/Genso.Astrology#3] [General]
When analyzing the planet positions. I see they are atleast 1deg 27min forward than what I see in Jaganatha Hora (with True Lahari/Chitrapaksha aynamsa)

I guess you are using Raman Aynamsa. And I believe this affects Dasa calculations. 

Is there a way to select Aynamsa ?

Also due to this when Jupiter in 23.37 deg in Sagi, reads as next house Capricorn for analysis and data

**Comment by sengiv** [comment]
Thank you very much for bringing this to our attention &
also for posting the 1st issue in this repo. :clinking_glasses:

---

>  I see they are at least 1deg 27min forward than what I see in Jaganatha Hora (with True Lahari/Chitrapaksha ayanamsa)

Yes, this was done purposely, because we **_observed wrong predictions in the real world_** with the standard calculations used by everybody else. In attempt to solve this inaccuracy, it was devised to round the planet longitudes where needed. B.V. Raman mentioned rounding values to compensate for small inaccuracies in his [books](https://github.com/gen-so/VedAstro/tree/c6f0bd3e238ff6521cdac4721f5cdd1d811abd97/Books). Also when making final predictions, we always use many planets to conclude, never just 1. For these reasons, the rounding was implemented. At the design level, this is considered a _**temporary hack**_ and we are working to improve it.

Our goal here is to **create perfect real world predictions**, and not just follow standard practice regardless of real world result.
As such we are constantly looking into ways of improving the original calculations. If you have any ideas on this, please feel free to share. Perhaps can start a [discussion](https://github.com/gen-so/VedAstro/discussions) on this topic.

---
> Also due to this when Jupiter in 23.37 deg in Sagi, reads as next house Capricorn for analysis and data

This is a very interesting point, the best way I can clarify this is by asking the question below.
**_Is the prediction made for Jupiter in Capricorn completely wrong?_**

You will notice that as a planet nears the end of a sign it's nature & strength changes.
It is not the same as being in center of the sign. In this case, if Jupiter was near the center of Sagittarius.
Then the rounding would not change anything, and predictions made for Jupiter in Sagittarius will be very accurate.
But as the planet nears the end of the sign/house, it is not the same. And the prediction starts to slip.
It is also [documented in the code here](https://github.com/gen-so/VedAstro/blob/c6f0bd3e238ff6521cdac4721f5cdd1d811abd97/Library/managers/AstronomicalCalculator.cs#L2570).

If your findings are otherwise, please share your experience as it will help improve the app.

-------
> I guess you are using Raman Ayanamsa. And I believe this affects Dasa calculations.

Yes, the calculation used to correct the movement of the zodiac is from BV Raman's books. On a side note regarding the Dasa
calculation, recently we proved that the maths behind the [Dasa chart](https://www.vedastro.org/Dasa) is near perfect. You can observe this for yourself, by generating charts for famous characters with marked life events. Some events like "death" can be predicted with the existing algorithm to an accuracy of weeks. Which further proves the validity of B. V. Raman's calculations.

Below is small quote from B.V Raman on Ayanamsa that might be of interest you.
> The exact period when both the zodiacs coincided in the first point is not definitely known and accordingly the Ayanamsa-the precessional distance -varies from 19° to 23° . The star which marked the first point seems to have somehow disappeared though some believe that it is 11' east of the star Pisces. A number of dates is given as the year of the coincidence, viz., 361 A.D., 498 A.D., 394 A.D., 397 A.D., 559 A.D., etc.; which to accept, and which to reject, has been a matter of considerable doubt. No definite proof is available in favour of any one of the dates given above. **_No amount of mere speculation would be of any use, especially in such matters. Some attribute these differences to the supposed errors in Hindu observations._** Whatever they may be, it is not our purpose here to enter into any sort of discussion which would be purely of academical interest...

---
> Is there a way to select Aynamsa ?

This is definitely possible, it can be done with code almost instantly. 
But the GUI will take some time, so if there is enough support for this feature, we will implement it.
[Here is the current method.](https://github.com/gen-so/VedAstro/blob/c6f0bd3e238ff6521cdac4721f5cdd1d811abd97/Library/managers/AstronomicalCalculator.cs#L35)

**Comment by HemKal** [comment]
Feel free to move this to discussion 

However planet with 7deg motion left in one house should not be read as next house. Especially for slow moving planets. 
For fast moving planets that might be applicable.

**Reply by sengiv** [reply]
**7 degrees out of place does raise a red flag.**

I'd be quick to mark it as a bug, but after little testing*,
it seems only certain planets go off by more than 5 degrees, while other planets align
perfectly with classical Hindu panchaga (ephemeris).


This means there are 3 possibilities for this discrepancy.

Possibility 1 : Calculations provided by Swiss Ephemeris using NASA JPL data is wrong.
Possibility 2 : Classical calculations are wrong or have shifted in past 50 - 100 years (as stars do in reality), and no body updated it.
Possibility 3 : Bug in the code that consumes Swiss Ephemeris.


*will post testing data soon, with planets names

**Reply by HemKal** [reply]
Awesome. Thx for your great work

**Reply by sengiv** [reply]
@HemKal :pray: thank you for your support.

**Comment by sengiv** [comment]
Completely Fixed! ✅
Now VedAstro matches Jaganath Hora exactly. 
![image](https://github.com/VedAstro/VedAstro/assets/43817262/592578c1-7470-43fd-9996-82f180f64158)
